### PR TITLE
doc(readme): Update readme to show recommended nuget versions

### LINKED ISSUE #1075: Show recommended NuGet packages and versions in the README
As an OSS best practice, I'd like to know which release and pre-release package versions are recommended.

I've seen this practice used broadly in other OSS repos producing NuGet packages such as:
https://github.com/windows-toolkit/WindowsCommunityToolkit#nuget-packages

### PR DESCRIPTION
**Pull request by abhipsaMisra**
We have customers who look for information regrading the recommended nuget packages published from our repository. This PR adds the [shields.io](https://shields.io/) version badge for each nuget package that we publish.

You can view the updated doc here: https://github.com/Azure/azure-iot-sdk-csharp/tree/abmisr/nugetVersion

Fixes #1075

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -7,6 +7,31 @@ This repository contains the following:
 - **Microsoft Azure Provisioning device SDK for C#** to provision devices to Azure IoT Hub with .NET
 - **Microsoft Azure Provisioning service SDK for C#** to manage your Provisioning service instance from a back-end .NET application
 
+### Build status
+
+Due to security considerations, build logs are not publicly available.
```
Comment (vinagesh) [inline readme.md:12]: Should we showing this publicly? Is the Azure pipelines visible to everyone?

File: readme.md [hunk 2]
```diff
@@ -7,6 +7,31 @@ This repository contains the following:
...
+
+| Service Environment                                                   | Status |
+| ---                                                                   | ---    |
+| [Master](https://github.com/Azure/azure-iot-sdk-csharp/tree/master)   | [![Build Status](https://azure-iot-sdks.visualstudio.com/azure-iot-sdks/_apis/build/status/csharp/csharp-westus1?branchName=master)](https://azure-iot-sdks.visualstudio.com/azure-iot-sdks/_build?definitionId=44&_a=summary&repositoryFilter=9&branchFilter=23)|
+| [Preview](https://github.com/Azure/azure-iot-sdk-csharp/tree/preview) | [![Build Status](https://azure-iot-sdks.visualstudio.com/azure-iot-sdks/_apis/build/status/csharp/csharp-canary1?branchName=preview)](https://azure-iot-sdks.visualstudio.com/azure-iot-sdks/_build?definitionId=43&_a=summary&repositoryFilter=9&branchFilter=72)|
+
+### Recommended NuGet packages
```
Comment (vinagesh) [inline readme.md:19]: Great addition!

File: readme.md (hunk 1 above)
Comment (abhipsaMisra) [reply readme.md:12]: The pipeline details/ logs are not visible publicly, you need an AzureDevOps account, and appropriate permissions. All this shows is the last build status for our prod and canary pipelines.

File: readme.md [hunk 3]
```diff
@@ -7,6 +7,31 @@ This repository contains the following:
...
+
+| Package Name                                          | Release Version                                           | Pre-release Version   |
+| ---                                                   | ---                                                       | ---                   |
+| Microsoft.Azure.Devices.Client                        | [![NuGet][iothub-device-release]][iothub-device-nuget]    | [![NuGet][iothub-device-prerelease]][iothub-device-nuget]     |
```
Comment (vinagesh) [inline readme.md:23]: Should we also add the license badge for these packages?

File: readme.md (hunk 1 above)
Comment (vinagesh) [reply readme.md:12]: I get that. Was just wondering if this is something of value in the public readme.

File: readme.md (hunk 3 above)
Comment (abhipsaMisra) [reply readme.md:23]: Good idea, I can add one right next to the `Microsoft Azure IoT SDK for .NET` header, what do you think?

File: readme.md (hunk 1 above)
Comment (timtay-microsoft) [reply readme.md:12]: It's a sort of motivating tool we use to strive for green builds. It's embarrassing when this badge is red all the time

Comment (vinagesh) [reply readme.md:12]: ha ha I was just thinking of the embarrassing part and hence the question :)

File: readme.md (hunk 3 above)
Comment (vinagesh) [reply readme.md:23]: That should be good

**Comment by timtay-microsoft** [comment]
Should the pre-release column exclude the libraries that don't have a preview release yet?

File: readme.md (hunk 1 above)
Comment (abhipsaMisra) [reply readme.md:12]: Ah, got it! This is indicative of the stability of our code, it's pretty common for OSS projects to display this kind of info.

**Comment by abhipsaMisra** [comment]
> Should the pre-release column exclude the libraries that don't have a preview release yet?

I didn't want to add that since we'll have to add it back in when we do release a "preview" version. The only place where I've excluded packages is for "preview" packages which don't have a GA version available.

File: readme.md (hunk 3 above)
Comment (abhipsaMisra) [reply readme.md:23]: The license badge is hitting some issues with GitHub's rate limits. The workaround is to install a "Shields GitHub application", which requires some additional investigation. I'll hold off on adding this change in this PR.

**Pull request merged**